core/src/garden.ts

```diff
@@ -1522,10 +1522,16 @@ export class Garden {
     }
 
     const context = new TemplatableConfigContext(this, config)
-    // Inject action's variables
-    context.variables = context.var = {
-      ...context.variables,
-      ...config.variables,
+    // Hack: deny variables contexts here, because those have not been fully resolved yet.
+    const deniedContexts = ["var", "variables"]
+    for (const deniedContext of deniedContexts) {
+      Object.defineProperty(context, "var", {
```

Comment:
Oh, found a small mistake here, otherwise we forbid `var` only. Maybe should add another test case for that
```suggestion
      Object.defineProperty(context, deniedContext, {
```